spec/queries/project/subscriptions_pledged_value_query_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+RSpec.describe Project::SubscriptionsPledgedValueQuery do
+  describe ".call" do
+    let(:subscriptions_pledged_value) { Project::SubscriptionsPledgedValueQuery.new(project) }
+
+    before { allow_any_instance_of(Project).to receive(:recurring?).and_return(true) }
+
+    subject { subscriptions_pledged_value.call }
+
+    context "when the project has registered subscriptions" do
+      context "with paid transactions" do
+        let(:standard_plan) { create(:plan, amount: 3000) }
+        let(:premium_plan) { create(:plan, amount: 10000) }
+        let(:project) { create(:project, plans: [premium_plan, standard_plan]) }
+        let(:subscritpion_for_standard_plan) { create(:subscription, :paid, plan: standard_plan, project: project) }
+        let(:subscritpion_for_premium_plan) { create(:subscription, :canceled, plan: premium_plan, project: project) }
+
+        before do
+          create(:transaction, status: :paid, subscription: subscritpion_for_standard_plan)
+          create(:transaction, status: :paid, subscription: subscritpion_for_premium_plan)
+          create(:transaction, status: :refunded, subscription: subscritpion_for_standard_plan)
+        end
+
+        it "returns the paid transactions formatted amount" do
+          expect(subject).to eq 130
+        end
+      end
+
+      context "without paid transactions" do
+        let(:plan) { create(:plan) }
+        let(:project) { create(:project, plans: [plan]) }
+        let(:subscription) { create(:subscription, :paid, plan: plan, project: project) }
+
+        before { create(:transaction, status: :pending_payment, subscription: subscription) }
+
+        it { is_expected.to be_zero }
+      end
+    end
+
+    context "when the project does not have plans" do
+      let(:project) { create(:project) }
+
+      it { is_expected.to be_zero }
+    end
+
+    context "when the project does not have subscriptions" do
+      let(:plan) { create(:plan) }
+      let(:project) { create(:project, plans: [plan]) }
```

Comment:
Eu acho que dá pra vc remover esses `let(:project) { create(:project) }` com um no início do describe e no lugar de criar um projecto com plano, vc pode fazer apenas `let(:plan) { create(:plan, project: project) }`, daí por default vc tem um projeto sem plano e quando o contexto pedir um plano vc so faz a associação.